### ISSUE TITLE
Broken system with Ubuntu 22.04 where Microsoft packages are partially replaced by Canonical packages

### ISSUE THREAD

**Issue by Robinsond76** [dotnet/runtime#73764]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

My issue seems to be similar to [this one](https://github.com/dotnet/aspnetcore/issues/38332) except I'm on linux (popOS 22.04).
Summing up the problem - when I attempted to use ```dotnet watch run``` today, I got the ```  * You intended to execute a .NET SDK command:
      No .NET SDKs were found.``` message.

When I attempt to install the dotnet sdk using ```sudo apt install dotnet-sdk-6.0``` I get ```dotnet-sdk-6.0 is already the newest version (6.0.400-1).```

When I do dotnet --info, I get:
```global.json file:
  Not found

Host:
  Version:      6.0.8
  Architecture: x64
  Commit:       55fb7ef977

.NET SDKs installed:
  No SDKs were found.

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.8 [/usr/lib/dotnet/dotnet6-6.0.108/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.8 [/usr/lib/dotnet/dotnet6-6.0.108/shared/Microsoft.NETCore.App]
```
I've tried doing ```whereis dotnet``` and adding all those paths to my $PATH but it didn't solve my problem. It was working just fine yesterday. Today when I got on the computer, I ran system and vscode updates - I don't know if that caused the problem.

Here is my ```whereis dotnet```:
```/
usr/bin/dotnet /usr/lib/dotnet /etc/dotnet /usr/share/dotnet /usr/lib/dotnet/dotnet6-6.0.108/dotnet /usr/share/man/man1/dotnet.1.gz
```
Here is my path:
```
/home/robinson/.nvm/versions/node/v18.7.0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/robinson/.dotnet/tools
```

I've also tried removing all dotnet packages that showed up in ```apt list installed 'dotnet*'```, restarting and reinstalling.

Any help is appreciated.

### .NET Version

6.0

**Comment by vitek-karas** [comment]
It should look for SDK next to the `dotnet` executable. So based on the output - your setup picks `dotnet` in `/usr/lib/dotnet/dotnet6-6.0.108` and there is probably no SDK installed there. I don't know what would put it there (standard package installers should put it into `/usr/share/dotnet`.

**Comment by remcodewit83** [comment]
I had exactly the same issue yesterday, following the same steps you did there. Eventually remove the packages again and used the dotnet-install.sh script to install it and that works (instructions can be found here: https://www.how2shout.com/linux/how-to-install-net-core-dotnet-on-ubuntu-22-04-lts/.

**Comment by Robinsond76** [comment]
Hello everyone,
it seems people started having the same issue and someone posted some steps on how to solve it on my post on StackOverflow. [Here is the link.](https://stackoverflow.com/questions/73312785/dotnet-sdk-is-installed-but-not-recognized-linux-ubuntu-popos-22-04/73315368#73315368)

So I ended up solving the issue as well by doing the following:
I uninstalled everything I could that was dotnet related with: `sudo apt remove 'dotnet*'`
Then `sudo apt update`
Then I followed the instructions [on this link](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/sdk-6.0.400-linux-x64-binaries) to unpack the binaries straight into the /opt/ folder. Once I added the necessary lines to PATH (according to the instructions in the link), everything worked as expected.

I would however recommend following the instructions in the StackOverflow post first as using the package manager might be the better solution.
Thank you everyone.

**Comment by directhex** [comment]
I can't reproduce this issue. Fresh install of popOS 22.04, installing the distro `dotnet-sdk-6.0` package, yields a found SDK in the correct place:

```
$ dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.108
 Commit:    4e3a463d2b

Runtime Environment:
 OS Name:     pop
 OS Version:  22.04
 OS Platform: Linux
 RID:         ubuntu.22.04-x64
 Base Path:   /usr/lib/dotnet/dotnet6-6.0.108/sdk/6.0.108/

global.json file:
  Not found

Host:
  Version:      6.0.8
  Architecture: x64
  Commit:       55fb7ef977

.NET SDKs installed:
  6.0.108 [/usr/lib/dotnet/dotnet6-6.0.108/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.8 [/usr/lib/dotnet/dotnet6-6.0.108/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.8 [/usr/lib/dotnet/dotnet6-6.0.108/shared/Microsoft.NETCore.App]

Download .NET:
  https://aka.ms/dotnet-download

Learn about .NET Runtimes and SDKs:
  https://aka.ms/dotnet/runtimes-sdk-info
```

**Comment by LakshanF** [comment]
The issue seems to be resolved now

**Comment by roger-oliver** [comment]
I'd like to add my experience today. Yeah, I've had the exact problem and solved removing dotnet and reinstalling using the "scripted install". No other installation process worked (all I tried, seems to install - quite quick - but no SDK)

`sudo apt remove 'dotnet*' --purge`
and I've followed the scripted install [https://docs.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#scripted-install]

After that, dotnet started to work.
